provider_sources/xyzservices-providers.json

```
@@ -373,5 +373,98 @@
             "max_zoom": 12,
             "name": "MapTiler.Terrain"
         }
+    },
+    "OrdnanceSurvey": {
+        "Road_3857": {
+            "url": "https://api.os.uk/maps/raster/v1/zxy/Road_3857/{z}/{x}/{y}.png?key={key}",
+            "html_attribution": "Contains OS data &copy Crown copyright and database right {year}",
+            "attribution": "Contains OS data (C) Crown copyright and database right {year}",
+            "key": "<A valid OS MapsAPI Key. Get a free key here - https://osdatahub.os.uk/>",
```

```suggestion
            "key": "<insert your valid OS MapsAPI Key. Get a free key here - https://osdatahub.os.uk/>",
```

these placeholders need to start with `"<insert your"` as we use this string to check if a token is required or not (maybe a bit clumsily but that's how it is done now).